### PR TITLE
Minor improvements to qubes-dom0-update

### PR DESCRIPTION
**Pull request by DemiMarie**
This uses bash regex matching instead of the `echo | grep` anti-pattern,
replaces an `==` with `=`, adds a missing `--`, and unsets a variable
that will be checked for being set later.  It also replaces some uses of
`[` with `[[`.

The same approach used here could be used to refactor
qubes-gpg-client-wrapper if it were worth the time, which it probably
isn’t unless qubes-gpg-client-wrapper will see future use.

### CODE REVIEW

File: dom0-updates/qubes-dom0-update [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-find_regex_in_args() {
-    local regex="${1}"
-    shift 1
-
-    for arg in "${@}"; do
-        if echo "${arg}" | grep -q -e "${regex}"; then
-            return 0
-        fi
+#!/bin/bash --
+set -euo pipefail
```
Comment (marmarek) [inline dom0-updates/qubes-dom0-update:2]: Both `set -e` and `set -o pipefail` breaks this in several ways. At the very least, there are several places that operate on `$?` to handle various exit codes (not only errors) - with `set -e` all that will stop working. But then, there is `tar` call that is supposed to pack files on best-effort basis (not all those files may be present on every system) - pipefail breaks this.

**Review by marmarek — CHANGES_REQUESTED** [review]
In it current shape, this PR makes `qubes-dom0-update` not functioning at all...

File: dom0-updates/qubes-dom0-update [hunk 2]
```diff
@@ -222,75 +220,77 @@ echo "Using $UPDATEVM as UpdateVM to download updates for Dom0; this may take so
 
 # qvm-run by default auto-starts the VM if not running
 readonly dom0_updates_dir=/var/lib/qubes/dom0-updates
-qvm-run --nogui -q -u root -- "$UPDATEVM" "mkdir -m 775 -p -- '$dom0_updates_dir'" || exit 1
-qvm-run --nogui -q -u root -- "$UPDATEVM" "chown -R -- user:user '$dom0_updates_dir'" || exit 1
-qvm-run --nogui -q -- "$UPDATEVM" "rm -rf -- '$dom0_updates_dir/etc' '$dom0_updates_dir/var/lib/rpm'" || exit 1
-(cd / && exec tar -c -- var/lib/rpm etc/yum/vars etc/yum.repos.d etc/yum.conf etc/dnf/dnf.conf etc/pki/rpm-gpg/RPM-GPG-KEY-* 2>/dev/null) |
```
Comment (marmarek) [inline dom0-updates/qubes-dom0-update:228]: You are reverting 37be9fbe610c67069ad4e59b595cc56a51da0514, I don't think it stopped being relevant (even if default install doesn't use it).

**Review by marmarek — CHANGES_REQUESTED** [review]

File: dom0-updates/qubes-dom0-update [hunk 3]
```diff
@@ -222,75 +220,81 @@ echo "Using $UPDATEVM as UpdateVM to download updates for Dom0; this may take so
...
-   qvm-run --nogui -q --pass-io -- "$UPDATEVM" "LC_MESSAGES=C tar -C '$dom0_updates_dir' -x &&
+qvm-run --no-gui -q -u root -- "$UPDATEVM" "mkdir -m 775 -p -- '$dom0_updates_dir'" || exit 1
+qvm-run --no-gui -q -u root -- "$UPDATEVM" "chown -R -- user:user '$dom0_updates_dir'" || exit 1
+qvm-run --no-gui -q -- "$UPDATEVM" "rm -rf -- '$dom0_updates_dir/etc' '$dom0_updates_dir/var/lib/rpm'" || exit 1
+dnf_conf=(var/lib/rpm etc/yum.repos.d etc/dnf/dnf.conf etc/pki/rpm-gpg/RPM-GPG-KEY-*)
```
Comment (marmarek) [inline dom0-updates/qubes-dom0-update:226]: This now fails because `etc/pki/rpm-gpg/RPM-GPG-KEY-*` isn't getting expanded.

**Review by marmarek — CHANGES_REQUESTED** [review]

File: dom0-updates/qubes-dom0-update (hunk 3 above)
Comment (marmarek) [reply dom0-updates/qubes-dom0-update:226]: Maybe include the whole dir?

**Comment by marmarek** [comment]
I don't see specific error, but something is still broken - packages are not downloaded (or not found at all): https://openqa.qubes-os.org/tests/56662#step/switch_pool/15

**Comment by DemiMarie** [comment]
> I don't see specific error, but something is still broken - packages are not downloaded (or not found at all): https://openqa.qubes-os.org/tests/56662#step/switch_pool/15

That is weird; I will deal with it later.

**Comment by marmarek** [comment]
This already conflicts, and will conflict even more after merging https://github.com/QubesOS/qubes-core-admin-linux/pull/116. Closing for now. _If_ you are interested in working on it, please rebase (after #116 gets merged), make it actually work, and reopen.

**Pull request closed without merging**